tests/unit/models/test_email_address.py

```diff
@@ -356,14 +357,25 @@ def test_email_address_delivery_state(clean_db):
     # Calling a transition will change state and set timestamp to update on commit
     ea.mark_sent()
     # An email was sent. Nothing more is known
-    assert ea.delivery_state.NORMAL
+    assert ea.delivery_state.SENT
     assert str(ea.delivery_state_at) == str(db.func.utcnow())
 
+    # mark_sent() can be called each time an email is sent
+    ea.mark_sent()
+
     # Recipient is known to be interacting with email (viewing or opening links)
     ea.mark_active()
     assert ea.delivery_state.ACTIVE
     assert str(ea.delivery_state_at) == str(db.func.utcnow())
 
+    # Active state cannot be downgraded to normal state
+    with pytest.raises(StateTransitionError):
+        ea.mark_sent()
+
+    # The proper protocol is to call mark_sent() only when it's available:
+    if ea.mark_sent.is_available:
+        ea.mark_sent()
```

Comment:
I think the check should be whether `mark_sent()` is available or not. The lines above this block calls `mark_sent()` so we know it's available.

Reply:
Isn't that what the check is doing?

Reply:
I'm adding an `assert False` to confirm that the branch isn't being called, as there is no test otherwise.

Reply:
Oh, "check" meaning `assert` instead of `if`. Changing it to `assert not ea.mark_sent.is_available`